### ISSUE TITLE
ReturnInVoidContext false positive in dynamic module class initializer with `define_method`

### ISSUE THREAD

**Issue by doits** [rubocop/rubocop#14062]
## False positive

When using `define_method` in a dynamic module class initializer I get a `Lint/ReturnInVoidContext: Do not return a value in initialize.`:

```ruby
class MyModule < Module
  def initialize
    define_method :my_method do
      return true if [true, false].sample # <-- Lint/ReturnInVoidContext

      false
    end
  end
end

class MyClass
  include MyModule.new
end

MyClass.new.my_method
```

The code above is simplified for brevity, in my real code the method names are dynamic based on arguments passed to the initializer.

I think `return` is doing what it is supposed to do there and only applied to the block/dynamically defined method, or am I mistaken?

I think the flagging got introduced in this PR: https://github.com/rubocop/rubocop/pull/13960

## RuboCop version

```
$ [bundle exec] rubocop -V
1.75.1 (using Parser 3.3.7.4, Prism 1.4.0, rubocop-ast 1.43.0, analyzing as Ruby 3.4, running on ruby 3.4.2) +server [arm64-darwin24]
  - rubocop-rails 2.31.0
  - rubocop-performance 1.25.0
```

**Comment by Earlopain** [comment]
Sorry, I missed your comment on the PR.

I guess it should be reverted? `define_method` can be aliased to any name/called from any other function and RuboCop can't detect that. Similar to https://github.com/rubocop/rubocop/pull/12959. Do you agree @koic?

**Comment by Earlopain** [comment]
I'll prepare a PR soon for this
